godtools/App/Features/Dashboard/Presentation/Dashboard/DashboardViewModel.swift

```diff
@@ -27,71 +28,56 @@ class ToolsMenuViewModel: ToolsMenuViewModelType {
-        
-        self.flowDelegate = flowDelegate
-        self.initialDataDownloader = initialDataDownloader
-        self.languageSettingsService = languageSettingsService
-        self.localizationServices = localizationServices
-        self.favoritingToolMessageCache = favoritingToolMessageCache
-        self.analytics = analytics
-        self.translationsRepository = translationsRepository
-        
-        self.disableOptInOnboardingBannerUseCase = disableOptInOnboardingBannerUseCase
-        self.getAllFavoritedToolsUseCase = getAllFavoritedToolsUseCase
-        self.getAllToolsUseCase = getAllToolsUseCase
-        self.getBannerImageUseCase = getBannerImageUseCase
-        self.getFeaturedLessonsUseCase = getFeaturedLessonsUseCase
-        self.getLanguageAvailabilityUseCase = getLanguageAvailabilityUseCase
-        self.getLessonsUseCase = getLessonsUseCase
-        self.getOptInOnboardingBannerEnabledUseCase = getOptInOnboardingBannerEnabledUseCase
-        self.getSettingsParallelLanguageUseCase = getSettingsParallelLanguageUseCase
-        self.getSettingsPrimaryLanguageUseCase = getSettingsPrimaryLanguageUseCase
-        self.getSpotlightToolsUseCase = getSpotlightToolsUseCase
-        self.getToolCategoriesUseCase = getToolCategoriesUseCase
-        self.getToolIsFavoritedUseCase = getToolIsFavoritedUseCase
-        self.removeToolFromFavoritesUseCase = removeToolFromFavoritesUseCase
-        self.toggleToolFavoritedUseCase = toggleToolFavoritedUseCase
-        
-        self.navTitleFont = fontService.getFont(size: 17, weight: .semibold)
-    }
-    
-    private func getFlowDelegate() -> FlowDelegate {
+    private var unwrappedFlowDelegate: FlowDelegate {
         guard let flowDelegate = self.flowDelegate else {
             assertionFailure("FlowDelegate should not be nil.")
             return self.flowDelegate!
         }
         return flowDelegate
     }
     
-    func lessonsWillAppear() -> LessonsViewModel {
-        return LessonsViewModel(
-            flowDelegate: getFlowDelegate(),
+    private var chooseLanguageButton: UIBarButtonItem?
+    var shouldShowLanguageSettingsBarButtonItemPublisher = CurrentValueSubject<(shouldShowButton: Bool, barButtonItem: UIBarButtonItem?), Never>((false, nil))
```

Comment:
Hey @rachaelblue this CurrentValueSubject doesn't have reason to change and can be immutable.